tests/test_helpers.py

```diff
@@ -145,10 +145,65 @@ def test_deserialize_valid_5(self) -> None:
         )
 
     def test_deserialize_valid_6(self) -> None:
-        self.assertEqual(
-            XsdDateTime.deserialize('2001-10-26T21:32:52.12679'),
-            datetime(year=2001, month=10, day=26, hour=21, minute=32, second=52, microsecond=12679, tzinfo=None)
-        )
+        """Test that less than 6 decimal places in the seconds field is padded to 6 and parsed correctly."""
+
+        # Use mock wrapping datetime so we can check that it was given a correctly padded string
+        with mock.patch('serializable.helpers.datetime', wraps=datetime) as datetime_mock:
+
+            # Make sure the parsed datetime is what we expect
+            self.assertEqual(
+                XsdDateTime.deserialize('2001-10-26T21:32:52.12679'),
+                datetime(year=2001, month=10, day=26, hour=21, minute=32, second=52, microsecond=126790, tzinfo=None)
+            )
+
+            # Make sure the string provided to fromisoformat was correctly padded (pre 3.11 it needs 0 or 6 decimals)
+            datetime_mock.fromisoformat.assert_called_with('2001-10-26T21:32:52.126790')
```

Comment:
Are you sure? I know we don't care about how its internally called, its there check that `datetime.fromisotimestamp` is never given a timestamp that pre-3.11 cannot parse, so the test can fail if given a timestamp that is incompatible with pre-3.11 even if its being tested on 3.11+ since otherwise if testing on 3.11+ `datetime.fromisotimestamp` will accept any number of decimals and the test will pass, even if it wouldn't on pre-3.11.

Reply:
i am certain.

Reply:
done via https://github.com/madpah/serializable/pull/138/commits/fac9e233fe38c0575ca4ec55566b9688ddf32bb5